core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/allowances/mareco_impl/AcceleratingSlopeCoast.java

```diff
@@ -47,6 +101,7 @@ private double computeV(PhysicsRollingStock rollingStock, double v1, double vf)
         // the lowest possible speed that will catch up with target speed at the end
         double wle = rollingStock.getRollingResistance(v1) * v1 * vf / (v1 - vf);
         double accelRatio = previousAccelerationEstimate / slopeAverageAcceleration;
+        accelRatio = Math.min(0, accelRatio);
```

Comment:
That seems like a sane change :D

Reply:
But it *shouldn't* be needed anymore, that was a leftover of our previous investigations. I'm tempted to replace it with an `assert accelRatio <= 0`.

Reply:
You can try that out ! Maybe it will spot some bugs